articles/node_lifecycle.md

```diff
@@ -27,224 +27,447 @@ Last Modified: {% if page.last_modified %}{{ page.last_modified }}{% else %}{{ p
+These event publications or service responses can fail.
+These are expected after a fully successful transition, however, and thus do not affect the underlying state machine.
 
-The node uses this to set up any resources it must hold throughout its life (irrespective of if it is active or inactive).
-As examples, such resources may include topic publications and subscriptions, memory that is held continuously, and initialising configuration parameters.
+In the event the backend throws and exception or error mid-transition, the backend will attempt to recover. 
+If the error is prior to a calling a user transition callback, the state will fallback (i.e., follow the `CallbackReturn::FAILURE`) path.
+The `change_state` process will end and any client will be responded to accordingly.
+For example and as outlined in [rclcpp::#1880](https://github.com/ros2/rclcpp/issues/1880), the `OnStateMachineTransitioned` event publication can fail.
+If this failure happens after the state machine update but prior to the user callback being called (e.g., `Unconfgured` → `Configuring` → publication fails), the state machine will be transitioned via the `CallbackReturn::FAILURE` (e.g., back to `Unconfgured` within this example).
 
-#### Valid transitions out of Configuring
+If the error occurs after a user callback has been called, the state machine will instead follow a `CallbackReturn::ERROR` pathway.
+Any client will be responded to accordingly.
 
-- If the `onConfigure` callback succeeds the node will transition to `Inactive`
-- If the `onConfigure` callback results in a failure code (TODO specific code) the node will transition back to `Unconfigured`.
-- If the `onConfigure` callback raises or results in any other result code the node will transition to `ErrorProcessing`
+#### User Transition Functions
+From the user perspective, transitions are completed in user transition functions.
+Each Transition state registers exactly one user transition function.
+This is to avoid needing to handle a more complicated register/deregister and ordering scheme for multiple callbacks (see [rclcpp::#2216](https://github.com/ros2/rclcpp/issues/2216)).
+If a user would like multiple callbacks, they are expected to set up their own mechanism for doing so.
 
-### Transition State: CleaningUp
+All user transition functions are overridable and contain the prior state as an argument.
+This reduces having to explicitly write all user transition functions (e.g., the `Activating` callback may often just be a `CallbackReturn::SUCCESS` as Managed Entities are automatically transitioned). 
+All user transition functions are expected to return control back to the Lifecycle backend of a [Return Values](#return-values) either through function return type (synchronous) or response deferral (through a shared handler).
```

Comment:
Are the words `Return Values` supposed to be there, or is some other noun intended to follow?

Reply:
I also am generally hesistant about whether we want to suggest implementation recommendations. We perhaps could end it with just "All user transition functions are expected to return control back to the Lifecycle backend"